File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/VB_TPS_BugTracker/TPS_BugTracker/GUILayer/Bugs/frmDetalleBug.vb
Partial Public Class frmDetalleBug
    Inherits Form

    Public Sub New()
        InitializeComponent()
    End Sub

    Friend Sub mostrarBug(ByVal oBugSelected As Bug)
        txtNroBug.Text = oBugSelected.id_bug.ToString()
        txtTitulo.Text = oBugSelected.titulo
        txtProducto.Text = oBugSelected.n_producto
        txtFechaAlta.Text = oBugSelected.fecha_alta.ToString()
        txtDescripcion.Text = oBugSelected.descripcion
        txtEstado.Text = oBugSelected.n_estado
        txtPrioridad.Text = oBugSelected.n_prioridad
        txtCriticidad.Text = oBugSelected.n_criticidad
        dgv_historiales.Rows.Clear()

        For Each oHistorialBug As HistorialBug In oBugSelected.historial
            dgv_historiales.Rows.Add(New String() {oHistorialBug.fecha.ToString("dd/MM/yyyy"), oHistorialBug.n_responsable, oHistorialBug.n_estado, oHistorialBug.n_asignado_a})
        Next
    End Sub

    Private Sub btnAceptar_Click(sender As Object, e As EventArgs) Handles btnAceptar.Click
        Me.Close()
    End Sub
End Class

File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/C#_TPS_BugTracker/TPS_BugTracker/GUILayer/Bugs/frmDetalleBug.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker.GUILayer.Bugs
{
    public partial class frmDetalleBug : Form
    {
        public frmDetalleBug()
        {
            InitializeComponent();
        }

        internal void mostrarBug(Bug oBugSelected)
        {
            txtNroBug.Text = oBugSelected.id_bug.ToString();
            txtTitulo.Text = oBugSelected.titulo;
            txtProducto.Text = oBugSelected.n_producto;
            txtFechaAlta.Text = oBugSelected.fecha_alta.ToString();
            txtDescripcion.Text = oBugSelected.descripcion;
            txtEstado.Text = oBugSelected.n_estado;
            txtPrioridad.Text = oBugSelected.n_prioridad;
            txtCriticidad.Text = oBugSelected.n_criticidad;

            dgv_historiales.Rows.Clear();
            foreach (HistorialBug oHistorialBug in oBugSelected.historial)
                dgv_historiales.Rows.Add(new string[] { oHistorialBug.fecha.ToString("dd/MM/yyyy"), oHistorialBug.n_responsable, oHistorialBug.n_estado, oHistorialBug.n_asignado_a });
        }

        private void btnAceptar_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}
